src/Quahog/Client.php

```diff
@@ -257,14 +261,34 @@ private function _sendCommand($command)
     /**
      * A wrapper to cleanly read a response from clamd.
      *
+     * @param bool $removeId
+     * @param string $readUntil
      * @return string
+     * @throws ConnectionException
      */
-    private function _receiveResponse()
+    private function _receiveResponse($removeId = false, $readUntil = "\n")
     {
-        $result = $this->_socket->read(4096);
+        $result = null;
+        $readUntilLen = strlen($readUntil);
+        do {
+            if ($this->_socket->selectRead($this->_timeout)) {
+                $rt = $this->_socket->read(4096, PHP_NORMAL_READ);
+                if ($rt == "") {
+                    break;
+                }
+                $result .= $rt;
+                if (strcmp(substr($result, 0 - $readUntilLen), $readUntil) == 0) {
+                    break;
+                }
+            } else {
+                throw new ConnectionException("Timeout waiting to read response");
```

Comment:
Im throwing an exception because previously it wouldn't timeout until the tcp connection finally died (if the connection was due to network connectivity)